### ISSUE TITLE
Subscribe method causes error in IE11

### ISSUE THREAD

**Issue by jirehnimes** [SocketCluster/socketcluster-client#138]
Import and creating client is okay for IE11 but when using the **subscribe** method, it causes this error:
![image](https://user-images.githubusercontent.com/10472490/94656879-3f96e080-0333-11eb-9f43-fbe4377316e9.png)

The application will stop into a white blank screen. It doesn't happen when in Chrome. See my implementation:
![image](https://user-images.githubusercontent.com/10472490/94657000-69e89e00-0333-11eb-8176-7df9ab045ba0.png)

**Details:**
**SocketCluster Client Version:** 16.0.1
**Vue 2**: 2.6.11

**Expected:** To work in IE11 similarly as in Chrome.
**Actual:** White blank screen in IE11 with syntax error. Working smoothly in Chrome.

**Comment by jondubois** [comment]
@jirehnimes I have not seen or heard about this error before.

- Do you have a handler (listener) for the socket `subscribe` event somewhere else in the code which has a syntax error in it?
- Is `channelData` a valid JSON string (could it be a JSON parsing `SyntaxError`)?

**Comment by jirehnimes** [comment]
@jondubois Thanks for responding immediately! 

- _Do you have a handler (listener) for the socket subscribe event somewhere else in the code which has a syntax error in it?_

That's the only code block with the listener for socket subscribe. It's a method within a Vue component.

- _Is channelData a valid JSON string (could it be a JSON parsing SyntaxError)?_

I did code also the content of `channelData` and it is a valid JSON string. 

I just want to clarify that the error happens only on the declaration of the subscribe channel because I commented the listener block or the loop after the subscribe channel declaration, the error persists on IE11. If I comment the subscribe channel declaration itself, the error will not occur.

**Comment by jondubois** [comment]
I can't see the line numbers in your stack trace. Based on what I can see, it does not look like an error from inside the `socketcluster-client` code. Could it be prototype pollution from a different library? Can you try running the same code in isolation without other libraries?

**Comment by jirehnimes** [comment]
I doubled check and the issue only occurs when `socket.subscribe` is used.
![image](https://user-images.githubusercontent.com/10472490/94725738-d644bb00-038e-11eb-80a2-cb7b79033e93.png)

**Comment by jirehnimes** [comment]
@jondubois I removed any implementation of socket in the component and declared the `subscribe` method to the `socket.ts` file and the error didn't occurred even when the said file is still imported in the Vue component.

socket.ts
![image](https://user-images.githubusercontent.com/10472490/94731128-da74d680-0396-11eb-9028-8b046a0bdf68.png)

Vue component
![image](https://user-images.githubusercontent.com/10472490/94731222-042dfd80-0397-11eb-8a21-b1d71c3fcfab.png)

If that's the case, I can't trigger the subscribe method by a method in a Vue component in IE11 even though it's working fine in Chrome.

**Comment by jirehnimes** [comment]
@jondubois Tried downgrading to v14.3.1, with the `socket.subscribe` in the method in Vue component error didn't occurred in IE11 and displayed the app as expected which was the original implementation. I think there's something off in v16 for IE11? 

Vue component with SocketCluster-client v14.3.1
![image](https://user-images.githubusercontent.com/10472490/94737388-558eba80-03a0-11eb-8f76-179a753738a4.png)

**Comment by jirehnimes** [comment]
@jondubois Sorry for a lot of replies. I decided to use SocketCluster client v14.3.1 with `protocolVersion: 1` in the SocketCluster server v16.0.1 since it is working now with my full implementation for both Chrome and IE11.

**Comment by jondubois** [comment]
@jirehnimes Glad you found a solution, I was going to suggest this approach with protocolVersion. I have not heard of this issue before so I'm thinking that it has something to do with your bundler. It could be that IE11 does not support some of the newer JavaScript constructs which SCv16 requires.

**Comment by jirehnimes** [comment]
@jondubois Yeah, thanks! Thanks also for responding! It's kinda tricky too and did a lot of comment and uncomment to verify which one causes the issue before I started this ticket. 
Probably with the bundler? I'm using Vue CLI with Vue 2 because Vue 3 doesn't support IE11 yet. Adjusted the browser list to `not ie <= 8` to make the app works in IE11. These things were before applying SCC. Even though I added `socketcluster-client` to `transpileDependencies` in `vue.config.js`, the issue still occurs.

**Comment by ahmetbozdoan** [comment]
I am having a similar error.

I could not run the server v16.0.1 version with SocketCluster client v14.3.1.

I am getting such an error

![image](https://user-images.githubusercontent.com/18261537/101877323-e05d1400-3b9e-11eb-8002-7cd0627380b4.png)

 
I changed the protocol version as follows

![image](https://user-images.githubusercontent.com/18261537/101877396-fcf94c00-3b9e-11eb-805f-b3db64cd8c1c.png)

Where am I doing wrong?
